index.js

```diff
@@ -1,8 +1,11 @@
 'use strict';
-module.exports = function toFastProperties(obj) {
-	function f() {}
-	f.prototype = obj;
-	new f();
-	return;
-	eval(obj);
-};
+module.exports = function toFastproperties(o) {
+	function Sub() {}
+	Sub.prototype = o;
+	var receiver = new Sub(); // create an instance
+	function ic() { return typeof receiver.foo; } // perform access
```

Comment:
@jdalton I lifted it up from the test suite that tests for the optimization a few months ago and cleaned up every bit that didn't break the optimization. Lemme find it real quick.